https://bugs.gentoo.org/907901
https://bugzilla.mozilla.org/show_bug.cgi?id=1836781
https://github.com/nss-dev/nss/commit/c07c4e073d95a25343cbf56b4a830a71e432869e

From c07c4e073d95a25343cbf56b4a830a71e432869e Mon Sep 17 00:00:00 2001
From: Natalia Kulatova <nkulatova@mozilla.com>
Date: Mon, 5 Jun 2023 16:09:58 +0000
Subject: [PATCH] Bug 1836781 - Disabling ASM C25519 for A but X86_64
 r=bbeurdouche,nss-reviewers

Differential Revision: https://phabricator.services.mozilla.com/D179969

--HG--
extra : moz-landing-system : lando
--- a/lib/freebl/Makefile
+++ b/lib/freebl/Makefile
@@ -568,7 +568,9 @@ ifneq ($(shell $(CC) -? 2>&1 >/dev/null </dev/null | sed -e 's/:.*//;1q'),lcc)
             HAVE_INT128_SUPPORT = 1
             DEFINES += -DHAVE_INT128_SUPPORT
     else ifeq (1,$(CC_IS_GCC))
-        SUPPORTS_VALE_CURVE25519 = 1
+        ifeq ($(CPU_ARCH),x86_64)
+            SUPPORTS_VALE_CURVE25519 = 1
+        endif
         ifneq (,$(filter 4.6 4.7 4.8 4.9,$(word 1,$(GCC_VERSION)).$(word 2,$(GCC_VERSION))))
             HAVE_INT128_SUPPORT = 1
             DEFINES += -DHAVE_INT128_SUPPORT
